### PR TITLE
Hw 9

### PR DESCRIPTION
**Pull request by Fedich09**

### CODE REVIEW

File: src/main/java/mate/controller/car/GetMyCurrentCarsController.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            .getInstance(CarService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long driverId = (Long) req.getSession().getAttribute(DRIVER_ID);
+        if (driverId == null || carService.get(driverId) == null) {
+            resp.sendRedirect("/login");
+            return;
+        }
```
Comment (Ketch92) [inline src/main/java/mate/controller/car/GetMyCurrentCarsController.java]: Do you think this if is needed?

File: src/main/java/mate/controller/driver/CreateNewDriverController.java [hunk 2]
```diff
@@ -24,8 +24,18 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp)
...
+        String driverPassword = req.getParameter("driver_password");
+        String repeatDriverPassword = req.getParameter("repeat_driver_password");
+        if (driverPassword.equals(repeatDriverPassword)) {
+            Driver driver = new Driver(driverName, licenseNumber);
+            driver.setLogin(driverLogin);
+            driver.setPassword(driverPassword);
+            driverService.create(driver);
+            resp.sendRedirect(req.getContextPath() + "/");
+        } else {
+            req.setAttribute("message", "Yore password and repeat password aren't the same.");
```
Comment (Ketch92) [inline src/main/java/mate/controller/driver/CreateNewDriverController.java]:
```suggestion
            req.setAttribute("message", "Your password and repeat password aren't the same.");
```

File: src/main/java/mate/web/filter/AuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        Long driverId = (Long) req.getSession().getAttribute(DRIVER_ID);
+        String url = req.getServletPath();
+        if (url.equals("/login") || url.equals("/drivers/create")) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+        if (driverId == null || driverService.get(driverId) == null) {
+            resp.sendRedirect("/login");
+            return;
+        }
```
Comment (Ketch92) [inline src/main/java/mate/web/filter/AuthenticationFilter.java]: Check common mistakes

**Review by Ketch92 — CHANGES_REQUESTED** [review]

**Review by Ketch92 — APPROVED** [review]

File: src/main/java/mate/security/impl/AuthenticationServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import mate.lib.Inject;
+import mate.lib.Service;
+import mate.model.Driver;
+import mate.security.AuthenticationService;
+import mate.service.DriverService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    DriverService driverService;
```
Comment (RomanDubovskyi) [inline src/main/java/mate/security/impl/AuthenticationServiceImpl.java]: what about access modifiers?

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]
don't forget to fix the points, mentioned in the chat

File: src/main/java/mate/web/filter/AuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.service.DriverService;
+
+public class AuthenticationFilter implements Filter {
+    private static final String DRIVER_ID = "driver_id";
+    private static final Injector injector = Injector.getInstance("mate");
+    private DriverService driverService = (DriverService) injector
```
Comment (RomanDubovskyi) [inline src/main/java/mate/web/filter/AuthenticationFilter.java]: do we use it anywhere in the class?

File: src/main/java/mate/web/filter/AuthenticationFilter.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+                         FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        Long driverId = (Long) req.getSession().getAttribute(DRIVER_ID);
+        String url = req.getServletPath();
+        if (url.equals("/login") || url.equals("/drivers/create")) {
```
Comment (RomanDubovskyi) [inline src/main/java/mate/web/filter/AuthenticationFilter.java]: wrote about this in the chat

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]
you didn't fix comments from chat, be more attentive

File: src/main/java/mate/web/filter/AuthenticationFilter.java (hunk 6 above)
Comment (mfedenko09) [reply src/main/java/mate/web/filter/AuthenticationFilter.java]: sorry, I was think you mean I shouldn't forget about point what mentioned in this chat

Comment (RomanDubovskyi) [reply src/main/java/mate/web/filter/AuthenticationFilter.java]: ok, next time i will use 'channel', thanks)

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**